tests/testthat/test-derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,183 @@
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 2: filter with first_cond ----
+test_that("derive_var_confirmation_flag Test 2: filter with first_cond", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "CR" &
+        AVALC.join == "CR",
+      order = vars(AVISITN),
+      filter = TRUE
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   NA_character_,
+    "1",      2,        "CR",   "Y",
+    "1",      3,        "CR",   NA_character_,
+    "1",      4,        "SD",   NA_character_,
+    "1",      5,        "NE",   NA_character_,
+    "2",      1,        "SD",   NA_character_,
+    "2",      2,        "PR",   NA_character_,
+    "2",      3,        "PD",   NA_character_,
+    "3",      1,        "SD",   NA_character_,
+    "4",      1,        "PR",   NA_character_,
+    "4",      2,        "PD",   NA_character_,
+    "4",      3,        "SD",   NA_character_,
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 3: filter with first_cond and summary function ----
+test_that("derive_var_confirmation_flag Test 3: filter with first_cond and summary function", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "PR" &
+        AVALC.join %in% c("CR", "PR"),
+      order = vars(AVISITN),
+      filter = count_vals(AVALC.join, "SD") <= 1,
+      false_value = "N"
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   "Y",
+    "1",      2,        "CR",   "N",
+    "1",      3,        "CR",   "N",
+    "1",      4,        "SD",   "N",
+    "1",      5,        "NE",   "N",
+    "2",      1,        "SD",   "N",
+    "2",      2,        "PR",   "N",
+    "2",      3,        "PD",   "N",
+    "3",      1,        "SD",   "N",
+    "4",      1,        "PR",   "N",
+    "4",      2,        "PD",   "N",
+    "4",      3,        "SD",   "N",
+    "4",      4,        "SD",   "N",
+    "4",      5,        "PR",   "N"
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 4: join_type = "all" ----
+test_that("derive_var_confirmation_flag, Test 4: join_type = 'all'", {
+  adae <- tribble(
+    ~USUBJID, ~ADY, ~ACOVFL, ~ADURN,
+    "1",        10, "N",          1,
+    "1",        21, "N",         50,
+    "1",        23, "Y",         14,
+    "1",        32, "N",         31,
```

Comment:
include a >30 duration AE 8 days before the covid AE to show that this wont be flagged

Reply:
Added